In [3]:
import pandas as pd
import numpy as np
from scipy.stats import chisquare

In [4]:
df = pd.read_csv("ab_data.csv")

In [28]:
df.head()

user_id                   timestamp      group landing_page  converted
0   851104  2017-01-21 22:11:48.556739    control     old_page          0
1   804228  2017-01-12 08:01:45.159739    control     old_page          0
2   661590  2017-01-11 16:55:06.154213  treatment     new_page          0
3   853541  2017-01-08 18:28:03.143765  treatment     new_page          0
4   864975  2017-01-21 01:52:26.210827    control     old_page          1

## Cleaning the Data

Selecting rows where only the desired conditions exist

In [6]:
df_1 = df.loc[(df.group == 'control') & (df.landing_page == 'old_page')]
df_2 = df.loc[(df.group == 'treatment') & (df.landing_page == 'new_page')]

In [7]:
df_1.count()

user_id         145274
timestamp       145274
group           145274
landing_page    145274
converted       145274
dtype: int64

In [8]:
df_2.count()

user_id         145311
timestamp       145311
group           145311
landing_page    145311
converted       145311
dtype: int64

In [9]:
df_clean = pd.concat([df_1, df_2])

In [10]:
df_clean.head()

user_id                   timestamp    group landing_page  converted
0   851104  2017-01-21 22:11:48.556739  control     old_page          0
1   804228  2017-01-12 08:01:45.159739  control     old_page          0
4   864975  2017-01-21 01:52:26.210827  control     old_page          1
5   936923  2017-01-10 15:20:49.083499  control     old_page          0
7   719014  2017-01-17 01:48:29.539573  control     old_page          0

In [11]:
df_clean.count()

user_id         290585
timestamp       290585
group           290585
landing_page    290585
converted       290585
dtype: int64

Null hypothesis is where the mean of users who converted is the same for both the old and new\
pages, where as the alternative hypothesis would be that mean conversion rate is higher for\
the new page

We make a contingency table

In [30]:
chisqt = pd.crosstab(df_clean.landing_page, df_clean.converted, margins=True)
print(chisqt)

converted          0      1     All
landing_page                       
new_page      128047  17264  145311
old_page      127785  17489  145274
All           255832  34753  290585


In [20]:
value = np.array([chisqt.iloc[0][0:5].values, chisqt.iloc[1][0:5].values])

In [21]:
from scipy.stats import chi2_contingency

Run chi-square test

In [26]:
print(chi2_contingency(value)[0:3])

(1.7203143230181923, 0.4230955827520224, 2)


Statistical value of 1.72, P-value of 0.42, 2 degrees of freedom